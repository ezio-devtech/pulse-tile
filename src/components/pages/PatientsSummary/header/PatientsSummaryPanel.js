import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash/fp';
import { get } from 'lodash';
import { lifecycle } from 'recompose';
import { Row } from 'react-bootstrap';
import { Col } from 'react-bootstrap';

import PTCustomInput from './PTCustomInput';
import { unmountOnBlur } from '../../../../utils/HOCs/unmount-on-blur.utils'
import { patientsSummaryConfig } from '../patients-summary.config';
import { themeConfigs } from '../../../../themes.config';
import { dashboardBeing } from '../../../../plugins.config';

import ExtraPatientsSummarySelectors from '../../../theme/components/ExtraPatientsSummarySelectors';

@lifecycle(unmountOnBlur)

export default class PatientsSummaryPanel extends PureComponent {
  static propTypes = {
    onCategorySelected: PropTypes.func.isRequired,
    onViewOfBoardsSelected: PropTypes.func.isRequired,
    selectedCategory: PropTypes.objectOf(PropTypes.bool).isRequired,
    selectedViewOfBoards: PropTypes.objectOf(PropTypes.bool).isRequired,
  };

  state = {
    selected: this.props.selectedCategory,
    selectedViewOptions: this.props.selectedViewOfBoards,
  };

  componentDidUpdate(prevProps, prevState) {
    if (!_.isEqual(prevState.selected, this.state.selected)) this.props.onCategorySelected(this.state.selected);
    if (!_.isEqual(prevState.selectedViewOptions, this.state.selectedViewOptions)) this.props.onViewOfBoardsSelected(this.state.selectedViewOptions);
  }

  toggleCheckbox = key => this.setState((prevState) => {
    const newValue = !_.get(['selected', key])(prevState);
    const keyName = 'isShow_' + key;
    localStorage.setItem(keyName, newValue);
    return _.set(['selected', key], newValue)(prevState);
  });

  toggleRadio = key => this.setState((prevState) => {
    const selectedViewOptions = Object.assign({}, prevState.selectedViewOptions);
    selectedViewOptions.full = false;
    selectedViewOptions.preview = false;
    selectedViewOptions.list = false;
    selectedViewOptions[key] = true;
    return { 'selectedViewOptions': selectedViewOptions };
  });

  render() {
    const { selected, selectedViewOptions } = this.state;
    const { patientsSummaryHasPreviewSettings, boards } = this.props;

    return (
      <div className="dropdown-menu dropdown-menu-panel dropdown-menu-summary">
        <div className="form-group-wrapper">
          <div className="heading">SHOW</div>
          <div className="form-group">
            <Row>
              {patientsSummaryConfig.map((item) => {
                return dashboardBeing[item.key] !== false ?
                  <Col xs={6} sm={4} key={item.nameCheckboxes}>
                    <PTCustomInput
                      type="checkbox"
                      title={item.titleCheckboxes}
                      id={item.nameCheckboxes}
                      name={item.nameCheckboxes}
                      isChecked={selected[item.key]}
                      onChange={this.toggleCheckbox}
                    />
                  </Col> : null
              })}
            </Row>
          </div>

          <ExtraPatientsSummarySelectors boards={boards} toggleCheckbox={this.toggleCheckbox} />

          {(themeConfigs.patientsSummaryHasPreviewSettings || patientsSummaryHasPreviewSettings) ?
            <div>
              <div className="heading">VIEW OF BOARDS</div>
              <div className="form-group">
                <Row>
                  <Col xs={12}>
                    <PTCustomInput type="radio" title="Headings + List" id="full" name="view-of-preview" value="full" isChecked={get(selectedViewOptions, 'full', null)} onChange={this.toggleRadio} />
                  </Col>
                </Row>
                <Row>
                  <Col xs={12}>
                    <PTCustomInput type="radio" title="Headings" id="preview" name="view-of-preview" value="preview" isChecked={get(selectedViewOptions, 'preview', null)} onChange={this.toggleRadio} />
                  </Col>
                </Row>
              </div>
            </div> : null}
        </div>
      </div>
    )
  }
}
