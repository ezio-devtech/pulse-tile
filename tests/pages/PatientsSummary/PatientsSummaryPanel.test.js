import React from 'react';
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-15';
configure({ adapter: new Adapter() });

import PatientsSummaryPanel from '../../../src/components/pages/PatientsSummary/header/PatientsSummaryPanel';

const testProps = {
	onCategorySelected: () => {},
	selectedCategory: {
		problems: true,
		contacts: true,
		allergies: true,
		medications: true,
	}
};

describe('Component <PatientsSummaryPanel />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
        <PatientsSummaryPanel
					onCategorySelected={testProps.onCategorySelected}
					selectedCategory={testProps.selectedCategory}
        />).dive();
		expect(component).toMatchSnapshot();

		expect(component.instance().props['onCategorySelected']).toEqual(testProps.onCategorySelected);
		expect(component.instance().props['selectedCategory']).toEqual(testProps.selectedCategory);

		expect(component.find('.heading')).toHaveLength(1);
		expect(component.find('.heading').text()).toEqual('SHOW');
		expect(component.find('.form-group')).toHaveLength(1);
		expect(component.find('PTCustomCheckbox')).toHaveLength(4);

		component.instance().toggleCheckbox('dashboard-name');
		component.setState({ selected: {
			problems: true,
			contacts: false,
			allergies: true,
			medications: false,
		}});
		component.setState({ selected: {
			problems: true,
			contacts: false,
			allergies: true,
			medications: false,
		}});
		expect(component.find('PTCustomCheckbox')).toHaveLength(4);
  });
});
