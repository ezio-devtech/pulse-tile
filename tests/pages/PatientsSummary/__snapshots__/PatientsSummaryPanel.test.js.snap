// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <PatientsSummaryPanel /> should renders with all props correctly 1`] = `
<div
  className="dropdown-menu dropdown-menu-panel dropdown-menu-summary"
>
  <div
    className="form-group-wrapper"
  >
    <div
      className="heading"
    >
      SHOW
    </div>
    <div
      className="form-group"
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <PTCustomCheckbox
          isChecked={true}
          name="problems"
          onChange={[Function]}
          title="Problems"
        />
        <PTCustomCheckbox
          isChecked={true}
          name="contacts"
          onChange={[Function]}
          title="Contacts"
        />
        <PTCustomCheckbox
          isChecked={true}
          name="allergies"
          onChange={[Function]}
          title="Allergies"
        />
        <PTCustomCheckbox
          isChecked={true}
          name="medications"
          onChange={[Function]}
          title="Medications"
        />
      </Row>
    </div>
  </div>
</div>
`;
